plugin/nsid/setup.go

```diff
@@ -0,0 +1,43 @@
+package nsid
+
+import (
+	"strings"
+
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+
+	"github.com/mholt/caddy"
+)
+
+func init() {
+	caddy.RegisterPlugin("nsid", caddy.Plugin{
+		ServerType: "dns",
+		Action:     setup,
+	})
+}
+
+func setup(c *caddy.Controller) error {
+	nsid, err := nsidParse(c)
+	if err != nil {
+		return plugin.Error("nsid", err)
+	}
+
+	dnsserver.GetConfig(c).AddPlugin(func(next plugin.Handler) plugin.Handler {
+		return Nsid{Next: next, Data: nsid}
+	})
+
+	return nil
+}
+
+func nsidParse(c *caddy.Controller) (string, error) {
+	nsid := "NSID"
```

Comment:
I believe the default was the hostname (I *think*), haven't read the RFC lately

Reply:
Thanks @miekg. Not familiar with this area. Now reading the RFC again, it does not mention the hostname. However, I did a google search and it turns out Bind is using hostname as the default (the related config seems to be `request-nsid`). I think it makes sense to conform to bind behavior. The PR has been updated.